src/sonic-yang-models/yang-models/sonic-fine-grained-ecmp.yang

```diff
@@ -0,0 +1,136 @@
+module sonic-fine-grained-ecmp {
+    
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-fine-grained-ecmp";
+    prefix sfgecmp;
+
+    import ietf-inet-types {
+        prefix inet;
+    }   
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC Fine Grained ECMP";
+
+    revision 2023-01-19 {
+        description "Initial revision.";
+    }
+
+
+    container sonic-fine-grained-ecmp {
+
+        container FG_NHG { 
+
+            description "FG_NHG part of config_db.json";
+
+            list FG_NHG_LIST{
+                key "name";
+
+                leaf name{
+                    type string;
+                    description "FG_NHG group name";
+                }
+
+                leaf bucket_size{
+                    type uint16;
+                    mandatory true;
+                    description "total hash bucket size desired, 
+                    recommended value of Lowest Common Multiple 
+                    of 1..{max # of next-hops}";
+                }
+
+                leaf match_mode{
+                    type string{
+                        pattern "(route-based|nexthop-based)"{
```

Comment:
Can we instead create a enumeration and make it a leaf ref to the enumeration?
Example: https://github.com/sonic-net/sonic-buildimage/blob/8f48773fd140e9fc27355427ae79386a5f5b7f9b/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L105

https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-vlan.yang#L243

Reply:
enum type match-mode-name added